tests/allschemes/channels008/Main.idr

```diff
@@ -0,0 +1,31 @@
+import System
+import System.Concurrency
+
+-- Simple producing thread.
+producer : Channel Nat -> Nat -> IO ()
+producer c n = ignore $ producer' n
+  where
+    producer' : Nat -> IO ()
+    producer' Z     = pure ()
+    producer' (S n) = do
+      channelPut c n
+      sleep 1
+
+-- Test that channelGetWithTimeout works as expected.
+main : IO ()
+main =
+  do c    <- makeChannel
+     tids <- for [0..11] $ \n => fork $ producer c n
+     vals <- for [0..11] $ \_ => channelGetWithTimeout c 5
+     ignore $ traverse (\t => threadWait t) tids
```

Comment:
lambda is redundant here: `ignore $ traverse threadWait tids`

Reply:
@emdash 

Due to the multiplicity on the `threadID` parameter of `threadWait`

```
threadWait : (1 threadID : ThreadID) -> IO ()
```

the original `ignore $ traverse (\t => threadWait t) tids` appears to be the correct way to do this, as `ignore $ traverse threadWait tids` throws:

```
Error: While processing right hand side of main. When unifying:
  (1 _ : ThreadID) -> IO ()
  and:
  ?a -> ?f ?b
Mismatch between: (1 _ : ThreadID) -> IO () and ?a -> ?f ?b.
```

Reply:
I stand corrected.